### DISCUSSION TITLE
Holes and coating/calibration + HA

### DISCUSSION THREAD

**Discussion by Nej4218** [rbaron/b-parasite#198] [Q&A]
I decided to go with BLE version of the parasites.
But i still have some final questions before ordering:

- Are the holes necessary or are they just for mounting the housing? 
![b-parasite-2 0 0~2](https://github.com/rbaron/b-parasite/assets/168338588/e78db51d-278f-47c2-9219-8879e97614e0)

- Clarification: the BOM is both for BLE & Zigbee? (The software/flashing is the only difference?)

- How do the sensors show up in home assistant? If they're on the dashboard could they not just be "calibrated" there? (Let's say you have the sensor in HA through ESPhome, you make a card and depending on voltage you make the card say "wet" if it's over the minimum voltage which you decide by placing it in soil you think is just about ready for water, if lower than that voltage make it say "dry")

- Has anyone set their devices up in a home assistant dashboard, if yes I'd love to see an example of someone's "cards" as I haven't been able to find any online 

- Is there a consensus on what coating you should use/works? Or is the standard just using conformal coating at jlcpcb and pray it holds up?

- If you keep the same length of the lines to the capacitor but make the board slightly shorter (like zigzags instead of straight line to achieve the same length of the mine going to the board) would it still in theory require recalibration of the sensor?
Imagine of idea for shortening the board slightly: 
![IMG_20240508_224915918~2](https://github.com/rbaron/b-parasite/assets/168338588/34326201-840d-44a1-ba45-b486ff69ca8f)

- also wondered about the line representing how deep the sensor should be, will i get completely different voltage readings if i put it in deeper (to just above the battery) / would this also require recalibration? Or is there a other reason for the line like avoiding risk of getting hardware wet?

Sorry about all these question I truly tried navigating the discussions/files and so forth to find an answer but without luck

**Comment by rbaron** [comment]
Hey @Nej4218,


> Are the holes necessary or are they just for mounting the housing?

They are just mounting holes.

> Clarification: the BOM is both for BLE & Zigbee? (The software/flashing is the only difference?)

Correct, single BOM for all sample firmwares.

> How do the sensors show up in home assistant? If they're on the dashboard could they not just be "calibrated" there? (Let's say you have the sensor in HA through ESPhome, you make a card and depending on voltage you make the card say "wet" if it's over the minimum voltage which you decide by placing it in soil you think is just about ready for water, if lower than that voltage make it say "dry")

> Has anyone set their devices up in a home assistant dashboard, if yes I'd love to see an example of someone's "cards" as I haven't been able to find any online

I don't have any custom cards for them, but I'd also be curious to see some fancy ones. I just pull the info from HA into a custom display. The default device view for the BTHome looks like this in my HA version:

<img width="672" alt="image" src="https://github.com/rbaron/b-parasite/assets/1573409/4b0a57c4-28b8-4ab6-ae23-48f90fc63d11">

About the calibration: there's two levels of calibration in your setup. The first one is the b-parasite board calibration. It sets the valid range from fully dry to fully wet. The second one is the threshold of what you would consider dry and wet (e.g.: 50% in that first calibration scale). We only do the first calibration; the second one is up to you and your type of plant, soil.

> Is there a consensus on what coating you should use/works? Or is the standard just using conformal coating at jlcpcb and pray it holds up?

See the [Conformal Coating wiki page](https://github.com/rbaron/b-parasite/wiki/Conformal-Coating) and #79.

> If you keep the same length of the lines to the capacitor but make the board slightly shorter (like zigzags instead of straight line to achieve the same length of the mine going to the board) would it still in theory require recalibration of the sensor?
Imagine of idea for shortening the board slightly:
IMG_20240508_224915918~2

The trace length to the capacitive pad is probably much less important than the shape of the pad itself. If you check out the [Hardware Versions](https://github.com/rbaron/b-parasite/wiki/Hardware-Versions), you'll see that the board length increased. That's because I had a lot of issues with water creeping in while watering the plants, when the water level briefly rises above the soil.
Also, it would be cool if you kept the attribution (see [license](https://github.com/rbaron/b-parasite?tab=readme-ov-file#license)).

> also wondered about the line representing how deep the sensor should be, will i get completely different voltage readings if i put it in deeper (to just above the battery) / would this also require recalibration? Or is there a other reason for the line like avoiding risk of getting hardware wet?

It's just a visual cue for calibration and keeping a consistent level when inserting it into the soil. It probably wouldn't require recalibration for the precision we aim for (see [What to Expect from the Sensor](https://github.com/rbaron/b-parasite/wiki/What-to-Expect-from-the-Sensor)).
